File: DotNet/TicTacToe.Lib/IntermediatePlayer.cs
using System;
using TicTacToeChallenge.Lib.DataClasses;

namespace TicTacToe.DotNet.Lib
{
    public class IntermediatePlayer : Player
    {
        public IntermediatePlayer() 
            : base("Intermediate")
        {

        }

        Random random = new Random();
        public override Cell Play(TicTacToeBoard board)
        {
            // Play randomly for now
            var randomCell = this.random.Next(board.AvailableCells.Count);
            var cell = board.AvailableCells[randomCell];
            Console.WriteLine("{0} player, playing at {1}", this.Name, cell.Name);
            return cell;
        }
    }
}

File: Python/IntermediatePlayer.py
def  IntermediatePlayer:    
    None
    public override Cell Play(TicTacToeBoard board)
    {
        // Play randomly for now until a Human changes this code to do something more specific
        var randomCell = board.GetRandomAvailableCell();
        Console.WriteLine("{0} player, playing at {1}", this.Name, randomCell.Name);
        return randomCell;
    }
